apps/api-extractor-model/src/model/ApiFunction.ts

```diff
@@ -63,4 +64,12 @@ export class ApiFunction extends ApiNameMixin(ApiTypeParameterListMixin(ApiParam
   public get containerKey(): string {
     return ApiFunction.getContainerKey(this.name, this.overloadIndex);
   }
+
+  /** @beta @override */
+  public buildCanonicalReference(): DeclarationReference {
+    return (this.parent ? this.parent.canonicalReference : DeclarationReference.empty())
+      .addNavigationStep(Navigation.Exports, this._getCanonicalReferenceName())
```

Comment:
Can an ApiFunction ever be created for a non-exported function that still may need to be referenced? This is valid TypeScript today:

```ts
// foo.d.ts
function f(): number; // not exported, ref should be `foo!~f:function(1)`
export function g(): typeof f; // exported, ref should be `foo!g:function(1)`
```

Reply:
Nope.  Currently `function f(): number;` appears in the .d.ts rollup but not in the .api.json file.